src/components/markdown/markdown.tsx

```diff
@@ -0,0 +1,55 @@
+import { Component, h, Prop, Watch } from '@stencil/core';
+import { markdownToHTML } from './markdown-parser';
+
+/**
+ * The Markdown component receives markdown syntax
+ * and renders it as HTML.
+ * @exampleComponent limel-example-markdown-headings
+ * @exampleComponent limel-example-markdown-emphasis
+ * @exampleComponent limel-example-markdown-lists
+ * @exampleComponent limel-example-markdown-links
+ * @exampleComponent limel-example-markdown-images
+ * @exampleComponent limel-example-markdown-code
+ * @exampleComponent limel-example-markdown-footnotes
+ * @exampleComponent limel-example-markdown-tables
+ * @exampleComponent limel-example-markdown-html
+ * @exampleComponent limel-example-markdown-blockquotes
+ * @exampleComponent limel-example-markdown-horizontal-rule
+ */
+@Component({
+    tag: 'limel-markdown',
+    styleUrl: 'markdown.scss',
+    shadow: true,
+})
+export class Markdown {
+    /**
+     * The input text. Treated as GitHub Flavored Markdown, with the addition
+     * that any included HTML will be parsed and rendered as HTML, rather than
+     * as text.
+     */
+    @Prop()
+    public content: string;
```

Comment:
There are both pros and cons for having this as a prop vs having the content provided as child elements instead. How did the discussion go about this topic when it was implemented?

Pros for having it as child elements is that it makes more sense, especially if HTML is allowed as well. Text is flowing content, and forcing the consumer to put it all in a string first is cumbersome. E.g.

```html
<limel-markdown>
    Hello, 
    <span class="green">World!</span>
</limel-markdown>
```
vs
```html
<limel-markdown 
    content="Hello, \n<span class=\"green\">World!</span>">
</limel-markdown>
```

Cons might be that it is a bit harder to implement for us.

If we insist on keeping this as a prop, I think we should make it more compatible with our other components and name this `value` instead of `content`

Reply:
> If we insist on keeping this as a prop, I think we should make it more compatible with our other components and name this value instead of content

Agree. Check this out, kind of relevant https://github.com/Lundalogik/crm-feature/issues/3480#issuecomment-1724922053

Reply:
If I recall correctly, we went with passing it as a prop because it allowed us to escape the linter/formatter that would automatically reformat the markdown text, which would then break the markdown. 
The only way to force the linter not to automatically reformat everything was a succession of ugly 'ignore' comments.

Because of that, we felt made more sense to pass it in as a prop.

Reply:
I still think we should rename the `content` prop to `value`

Reply:
will u do this @john-traas please? there seems to be some conflicts in the branch as well, that needs resolving.

Reply:
Yes, I am on it @Kiarokh

Reply:
It looks to me like this was resolved in favour of keeping it as a property, rather than a slot, but just in case:

I would assume that anytime someone wants to use this element, they have the content they need to present as a string, most likely from the database?

Also, I don't see why you'd ever need to put html into this component if you already have it as a DOM-structure? If it's for the sake of sanitising the html, that will already be too late by the time this element picks it up, as you've already injected the html into the DOM before the slot can "pick it up" for processing. (Or at least I'd presume it would be too late? And even if it wasn't, we'd presumably only be a small change to the `slot` functionality away from a security problem.)

Anyway, this might feel like I'm trying to hammer home that I think using `slot` is a bad idea, but it's more that I'm wondering if there's something I'm missing? 🤗

Reply:
I think we're on the same page, keeping it as a `property` is the approach we're going with.